pkg/inmemorychannel/event_dispatcher.go

```diff
@@ -68,10 +68,12 @@ func (d *InMemoryEventDispatcher) WaitReady() {
 }
 
 func NewEventDispatcher(args *InMemoryEventDispatcherArgs) *InMemoryEventDispatcher {
-	// TODO set read timeouts?
 	bindingsReceiver := kncloudevents.NewHTTPEventReceiver(
 		args.Port,
-		args.HTTPEventReceiverOptions...,
+		append(
+			args.HTTPEventReceiverOptions,
+			kncloudevents.WithReadTimeout(args.ReadTimeout),
```

Comment:
```suggestion
			kncloudevents.WithReadTimeout(args.ReadTimeout),
			args.HTTPEventReceiverOptions,
```

nit: can we switch those? Because then the users ReceiverOptions have priority and override our `kncloudevents.WithReadTimeout(args.ReadTimeout)`. For example if the user has not set the `args.ReadTimeout`, but has the `WithReadTimeout()` option defined on their own and added to `args.HTTPEventReceiverOptions`.

Reply:
Done !